components/toast/AppToast.vue

```diff
@@ -0,0 +1,271 @@
+<script>
+import {
+  defineComponent,
+  ref,
+  shallowRef,
+} from 'vue'
+
+import {
+  Icon,
+} from '#components'
+
+import useToastStore from '~/stores/toast'
+
+import AppToastContext from './AppToastContext'
+
+export default defineComponent({
+  components: {
+    Icon,
+  },
+
+  props: {
+    id: {
+      type: Symbol,
+      required: true,
+    },
+    title: {
+      type: [
+        String,
+        null,
+      ],
+      required: false,
+      default: null,
+    },
+    message: {
+      type: String,
+      required: true,
+    },
+    color: {
+      type: /** @type {import('vue').PropType<import('./AppToastContext').PropsType['color']>} */ (String),
+      required: false,
+      default: 'neutral',
+    },
+    timeout: {
+      type: Number,
+      required: false,
+      default: 5000,
+    },
+    shouldHideProgressBar: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    shouldPauseOnHover: {
+      type: Boolean,
+      required: false,
+      default: true,
+    },
+    iconName: {
+      type: String,
+      required: false,
+      default: '',
+    },
+  },
+
+  setup (
+    props,
+    componentContext
+  ) {
+    const toastStore = useToastStore()
+    const isRunningRef = ref(true)
```

Comment:
@hecticme 

empty line

```suggestion
    const toastStore = useToastStore()

    const isRunningRef = ref(true)
```